### ISSUE TITLE
Image quality gap between URAE and FLUX 1.1 Pro Ultra

### ISSUE THREAD

**Issue by xuyxu** [Huage001/URAE#3]
Hi,

Thanks for the great work showing that generating 2k*2k image with FLUX.1-Dev is not that hard.

I found that there is still a gap between URAE Lora and FLUX 1.1 Pro Ultra. For example, using the following prompt with sample steps 30 and guidance scale 3.5 to generate an image with size 2048*2048:

```
A serene scene of a river winding through a lush green forest, with sunlight filtering through the trees and casting gentle shadows on the water's surface. the river is surrounded by dense foliage, with rocks and trees on either side, and the sunlight filters through the leaves, creating a peaceful and tranquil atmosphere. the water is calm and clear, reflecting the sunlight and creating a sense of serenity and tranquility. the image is taken from a low angle, looking down on the river and the surrounding nature, with a focus on the natural beauty of the scene. the scene is peaceful and serene, with the river winding its way through the forest, surrounded by lush greenery and trees.
```

This is the result of FLUX 1.1 Pro Ultra:

![Image](https://github.com/user-attachments/assets/22c1072a-141b-44a9-a714-7b9676f0166f)

And this is the result of URAE 2K adapter:

![Image](https://github.com/user-attachments/assets/513c166a-73be-47fd-9a2d-1cfd65f19572)

One problem is that there are moire patterns on the image details of the URAE result:

<img width="563" alt="Image" src="https://github.com/user-attachments/assets/f862c82d-5cd3-4a20-843b-89474d72adc8" />

Any idea on how to alleviate this problem? Thanks!

**Comment by Lexie-YU** [comment]
Hi @xuyxu , thanks very much for your attention to our work. 
We follow your prompt and settings to generate image through our demo [https://huggingface.co/spaces/Yuanshi/URAE_dev](url) and the results are shown as follows:

![Image](https://github.com/user-attachments/assets/eacd07b9-d4ae-4bd0-9048-cd3bdf9d8db5)

We did not observe moiré patterns in our output.
Could you kindly share your code with us? This would greatly help us better understand the issue and assist you more effectively.

**Comment by xuyxu** [comment]
Sure, @Lexie-YU. Thanks for your help, here is my snippet:

```python
import os
import torch
from huggingface_hub import hf_hub_download

from pipeline_flux import FluxPipeline
from transformer_flux import FluxTransformer2DModel

bfl_repo = "black-forest-labs/FLUX.1-dev"
transformer = FluxTransformer2DModel.from_pretrained(bfl_repo, subfolder="transformer", torch_dtype=torch.bfloat16)
pipe = FluxPipeline.from_pretrained(bfl_repo, transformer=transformer, torch_dtype=torch.bfloat16)
pipe.enable_model_cpu_offload()  #save some VRAM by offloading the model to CPU. Remove this if you have enough GPU power

# Raw URAE
if not os.path.exists('ckpt/urae_2k_adapter.safetensors'):
    hf_hub_download(repo_id="Huage001/URAE", filename='urae_2k_adapter.safetensors', local_dir='ckpt', local_dir_use_symlinks=False)
pipe.load_lora_weights("ckpt/urae_2k_adapter.safetensors")

image = pipe(
    prompt="A serene scene of a river winding through a lush green forest, with sunlight filtering through the trees and casting gentle shadows on the water's surface. the river is surrounded by dense foliage, with rocks and trees on either side, and the sunlight filters through the leaves, creating a peaceful and tranquil atmosphere. the water is calm and clear, reflecting the sunlight and creating a sense of serenity and tranquility. the image is taken from a low angle, looking down on the river and the surrounding nature, with a focus on the natural beauty of the scene. the scene is peaceful and serene, with the river winding its way through the forest, surrounded by lush greenery and trees.",
    height=2048,
    width=2048,
    guidance_scale=3.5,
    num_inference_steps=30,
    max_sequence_length=512,
    generator=torch.Generator("cpu").manual_seed(0)
).images[0]

image.save("flux-urae-2k.png")
```

**Comment by Lexie-YU** [comment]
Hi, @xuyxu 
Thanks very much for your reply.
The issue can be resolved by adding 
`pipe.scheduler.config.use_dynamic_shifting = False`
`pipe.scheduler.config.time_shift = 10`
 to the code. 
We sincerely apologize for missing these lines in the README.md earlier, and we have now corrected it. Please check it.
Thanks for pointing it out.
For more details, please feel free to refer the inference demo file https://github.com/Huage001/URAE/blob/main/inference_2k.ipynb

**Comment by xuyxu** [comment]
WOW, solved. Thanks!
